scene/animation/animation_player.cpp

```diff
@@ -641,6 +697,49 @@ double AnimationPlayer::get_current_animation_length() const {
 	return playback.current.from->animation->get_length();
 }
 
+void AnimationPlayer::set_current_section_with_markers(const StringName &p_start_marker, const StringName &p_end_marker) {
+	ERR_FAIL_NULL_MSG(playback.current.from, "AnimationPlayer has no current animation.");
+	ERR_FAIL_COND_MSG(p_start_marker == p_end_marker && p_start_marker, vformat("Start marker and end marker cannot be the same marker: %s.", p_start_marker));
+	ERR_FAIL_COND_MSG(p_start_marker && !playback.current.from->animation->has_marker(p_start_marker), vformat("Marker %s not found in animation: %s.", p_start_marker, playback.current.from->animation->get_name()));
+	ERR_FAIL_COND_MSG(p_end_marker && !playback.current.from->animation->has_marker(p_end_marker), vformat("Marker %s not found in animation: %s.", p_end_marker, playback.current.from->animation->get_name()));
+	double start_time = p_start_marker ? playback.current.from->animation->get_marker_time(p_start_marker) : -1;
+	double end_time = p_end_marker ? playback.current.from->animation->get_marker_time(p_end_marker) : -1;
+	set_current_section(start_time, end_time);
```

Comment:
```suggestion
	double start_time = p_start_marker ? playback.current.from->animation->get_marker_time(p_start_marker) : -1;
	double end_time = p_end_marker ? playback.current.from->animation->get_marker_time(p_end_marker) : -1;
	ERR_FAIL_COND_MSG(p_start_marker && Animation::is_less_approx(start_time, 0), vformat("Marker %s time must be positive in animation: %s.", p_start_marker, playback.current.from->animation->get_name()));
	ERR_FAIL_COND_MSG(p_end_marker && Animation::is_less_approx(end_time, 0), vformat("Marker %s time must be positive in animation: %s.", p_end_marker, playback.current.from->animation->get_name()));
	set_current_section(start_time, end_time);
```
Negative time are invalid and will be at both ends of the animation, so if a marker is at a negative value, an error or warning is required to make it positive.

Reply:
Negative time markers can be placed in the UI and follows the rules of `play_section` (so any negative marker gets treated as empty markers), so I think this should be a warning instead.